bpf/kmesh/workload/include/authz.h

```diff
@@ -182,13 +187,300 @@ static int match_dst_ports(Istio__Security__Match *match, struct xdp_info *info,
     return UNMATCHED;
 }
 
+static inline __u32 convert_ipv4_to_u32(const struct ProtobufCBinaryData *ipv4_data)
+{
+    if (!ipv4_data->data || ipv4_data->len != 4) {
+        return 0;
+    }
+
+    unsigned char *data = (unsigned char *)KMESH_GET_PTR_VAL(ipv4_data->data, unsigned char);
+    if (!data) {
+        BPF_LOG(INFO, AUTH, "convert_ipv4_to_u32: Failed to read data from ipv4_data\n");
+        return 0;
+    }
+
+    return (data[3] << 24) | (data[2] << 16) | (data[1] << 8) | (data[0] << 0);
+}
+
+static inline __u32 convert_ipv6_to_u32(struct ip_addr *rule_addr, const struct ProtobufCBinaryData *ipv6_data)
```

Comment:
is u32 bige enough to hold ipv6？

Reply:
> is u32 bige enough to hold ipv6？

this function name is a misunderstanding it is an u32 array, I have changed the name